core/trino-main/src/test/java/io/trino/operator/TestDistinctLimitOperator.java

```diff
@@ -168,8 +169,8 @@ public void testMemoryReservationYield(Type type)
                 blockTypeOperators);
 
         GroupByHashYieldAssertion.GroupByHashYieldResult result = finishOperatorWithYieldingGroupByHash(input, type, operatorFactory, operator -> ((DistinctLimitOperator) operator).getCapacity(), 1_400_000);
-        assertGreaterThan(result.getYieldCount(), 5);
-        assertGreaterThan(result.getMaxReservedBytes(), 20L << 20);
+        assertGreaterThan(result.getYieldCount(), 4);
+        assertGreaterThanOrEqual(result.getMaxReservedBytes(), 20L << 20);
```

Comment:
why?

Reply:
Please add a note to commit message about test changes

Reply:
done

Reply:
I still don't understand why yield count decreased. Looking at `finishOperatorWithYieldingGroupByHash` there is:
```
            // saturate the pool with a tiny memory left
            long reservedMemoryInBytes = memoryPool.getFreeBytes() - additionalMemoryInBytes;
            memoryPool.reserve(anotherTaskId, "test", reservedMemoryInBytes);
```
so it should saturate pool regardless of memory consumption per row. It should only matter that hash table needs to be rehashed. Maybe
```
            // Skip if the memory usage is not large enough since we cannot distinguish
            // between rehash and memory used by aggregator
            if (newMemoryUsage < DataSize.of(4, MEGABYTE).toBytes()) {
```
4MB is too large.